src/Extension.ml

```diff
@@ -21,21 +21,75 @@ module Server = struct
     { command; args; options = { env = Process.env } }
 end
 
+module Instance = struct
+  type t = LanguageClient.t option ref
+
+  let create () = ref None
+
+  let stop t =
+    match !t with
+    | None -> ()
+    | Some (client : LanguageClient.t) ->
+      (client.stop () [@bs]);
+      t := None
+
+  let start t toolchain =
+    let open Promise.Result.O in
+    Toolchain.runSetup toolchain >>| fun () ->
+    let serverOptions = Server.make toolchain in
+    let client =
+      LanguageClient.make ~id:"ocaml" ~name:"OCaml Language Server"
+        ~serverOptions ~clientOptions:(Client.make ())
+    in
+    t := Some client;
+    (client.start () [@bs])
+end
+
+let workspaceRoot () = Path.ofString Workspace.rootPath
+
+let selectSandbox (instance : Instance.t) () =
+  let setToolchain =
+    let open Promise.O in
+    let projectRoot = workspaceRoot () in
+    Toolchain.selectAndSave ~projectRoot >>= function
+    | None -> Promise.Result.return ()
+    | Some t ->
+      Instance.stop instance;
+      let t = Toolchain.makeResources ~projectRoot t in
+      Instance.start instance t
+  in
+  let (_ : unit Promise.t) = handleError Window.showErrorMessage setToolchain in
+  ()
+
+let suggestToSetupToolchain instance =
+  let open Promise.O in
+  Vscode.Window.showInformationMessage'
+    "There is no toolchain attached to this project."
+    [ ("select toolchain", ()) ]
```

Comment:
@prometheansacrifice This is what doesn't seem to be working at the moment. The function should suggest to setup a toolchain, but the "select toolchain" option doesn't show up.